spmd/tensor/parallel/utils.py

```diff
@@ -0,0 +1,51 @@
+import functools
+
+import torch
+from spmd.tensor import DeviceMesh, DTensor
+from typing import Callable, Optional, Union
+
+
+def _prepare_output_validate(
+    _prepare_output_func: Callable[
+        [DTensor, Optional[DeviceMesh], Optional[int]],
+        Union[torch.Tensor, DTensor],
+    ]
+) -> Callable[
+    [DTensor, Optional[DeviceMesh], Optional[int]], Union[torch.Tensor, DTensor]
+]:
+    """
+    Inject common validation logics for _prepare_output funcs via this 
+    decorator, including verifying that output needs to be a DTensor 
+    and only 1D Device Mesh is passed in.
+    Example::
+        >>> @_prepare_output_validate
+        >>> def make_output_shard_1d(args, kwargs):
+        >>>   ...
+        >>>
+        >>> dt = distribute(tensor, device_mesh, [Shard(0)])
+        >>> make_output_shard_1d(dt, device_mesh, 1)
+        >>> # This will call '_prepare_output_validate' first
+    Args:
+        _prepare_output_func (Callable): The func we want to inject the 
+            validation into.
+    Return:
+        func (Callable): Same input func with validation logic added.
+    """
+
+    @functools.wraps(_prepare_output_func)
+    def wrapper(*args, **kwargs):  # pyre-ignore[2, 3]
+        assert len(args) >= 2, "_prepare_output need at least two args."
```

Comment:
Question:
If the call is `make_output_replicate(tensor)` (i.e. device_mesh argument passed is None), will `len(args)` be just 1?

Reply:
yep.

Reply:
if we need at least two args, let's just make device mesh be non-optional?

Reply:
Opps, I thought it in a wrong way. We do want to make DeviceMesh Optional.. I will change the code.